mlfinlab/backtest_statistics/statistics.py

```diff
@@ -152,62 +152,96 @@ def all_bets_concentration(returns: pd.Series, frequency: str = 'M') -> tuple:
+    time_under_water = np.append(time_under_water,
+                                 (returns.index[-1] - high_watermarks.index[-1]) /
+                                 np.timedelta64(1, 'Y'))
+
+    time_under_water = pd.Series(time_under_water, index=high_watermarks.index)
     return drawdown, time_under_water
 
 
-def sharpe_ratio(returns: pd.Series, cumulative: bool = False,
-                 entries_per_year: int = 252, risk_free_rate: float = 0) -> float:
+def sharpe_ratio(returns: pd.Series, entries_per_year: int = 252,
+                 risk_free_rate: float = 0) -> float:
     """
-    Calculates Annualized Sharpe Ratio for pd.Series of normal (not log) returns.
+    Calculates Annualized Sharpe Ratio for pd.Series of normal or log returns.
+    Risk_free_rate should be given for the same period the returns are given.
+    For example, if the input returns are observed in 3 months, the risk-free
+    rate given should be the 3-month risk-free rate.
 
-    :param returns: (pd.Series) returns
-    :param cumulative: (bool) flag if returns are cumulative (no by default)
-    :param entries_per_year: (int) times returns are recorded per year (days by default)
+    :param returns: (pd.Series) returns - normal or log
+    :param entries_per_year: (int) times returns are recorded per year (daily by default)
     :param risk_free_rate: (float) risk-free rate (0 by default)
     :return: (float) Annualized Sharpe Ratio
     """
-    if cumulative:
-        returns = returns / returns.shift(1) - 1  # Inverting cumulative returns
-        returns = returns[1:]  # Excluding empty value
+
     sharpe_r = (returns.mean() - risk_free_rate) / returns.std() * \
                (entries_per_year) ** (1 / 2)
 
     return sharpe_r
 
 
-def probabalistic_sharpe_ratio(observed_sr: float, benchmark_sr: float,
+def information_ratio(returns: pd.Series, benchmark: float = 0,
+                      entries_per_year: int = 252) -> float:
+    """
+    Calculates Annualized Information Ratio for pd.Series of normal or log returns.
+    Benchmark should be provided as a return for the same time period as that between
+    input returns. For example, for the daily observations it should be the
+    benchmark of daily returns.
+
+    It is the annualized ratio between the average excess return and the tracking error.
+    The excess return is measured as the portfolio’s return in excess of the benchmark’s
+    return. The tracking error is estimated as the standard deviation of the excess returns.
+
+    :param returns: (pd.Series) returns - normal or log
+    :param benchmark: (float) benchmark for performance comparison (0 by default)
+    :param entries_per_year: (int) times returns are recorded per year (daily by default)
+    :return: (float) Annualized Information Ratio
+    """
+    excess_returns = returns - benchmark
+    information_r = sharpe_ratio(excess_returns, entries_per_year)
```

Comment:
Is this right? your calling sharpe ratio.

Reply:
Yes, the calculation is the same, except that it's excess returns over a benchmark instead of just returns.